File: backend/GymRecommender/models/request_period.py
from django.db import models
from .request import Request

class RequestPeriod(models.Model):
    id = models.UUIDField(primary_key=True)
    request = models.ForeignKey(Request, models.DO_NOTHING)
    weekday = models.IntegerField()
    arrival_time = models.TimeField(blank=True, null=True)
    departure_time = models.TimeField(blank=True, null=True)

    class Meta:
        managed = False
        db_table = 'request_period'
        unique_together = (('request', 'weekday'),)

File: backend/Models/RequestPeriod.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class RequestPeriod
{
    public Guid Id { get; set; }

    public Guid RequestId { get; set; }

    public int Weekday { get; set; }

    public TimeOnly? ArrivalTime { get; set; }

    public TimeOnly? DepartureTime { get; set; }

    public virtual Request Request { get; set; } = null!;
}
